src/api/domain/auth.api.client.ts

```diff
@@ -0,0 +1,28 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { AuthResponse } from '~/types/auth';
+import { generateCookiesKeyValues } from '~/utils/auth/tokenHandlers';
+
+import publicApi from '../config/publicApi';
+
+export const login = async (code: string | null) => {
+  const origin = window.location.origin;
+  const authData = await publicApi.post<AuthResponse>('/auth/login/kakao', {
+    authCode: code,
+    redirectUri: `${origin}/auth/callback/kakao`,
+  });
+  if (authData.data) {
+    const cookies = generateCookiesKeyValues(authData.data as AuthResponse);
+    cookies.forEach(([key, value]) => {
+      document.cookie = `${key}=${value}; path=/;`;
+    });
+  }
+
+  return authData;
+};
+
+export const useLogin = (code: string | null, options?: { onSuccess?: () => void }) => {
+  return useQuery(['login'], async () => login(code), {
+    ...(options ?? {}),
+  });
+};
```

Comment:
아래와 같이 query option type 을 사용해서 query option에 대해서 모두 받을 수 있도록 선언해 주는 건 어떨까요?
```suggestion
export const useLogin = (code: string | null, options?: UseQueryOptions<AuthResponse>) => {
  return useQuery<AuthResponse>(['login'], async () => login(code), options);
};
};
```